website_sale_checkout_country_vat/controllers/main.py

```diff
@@ -0,0 +1,57 @@
+# -*- coding: utf-8 -*-
+# © 2016 Sergio Teruel <sergio.teruel@tecnativa.com>
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+from openerp import http
+from openerp.http import request
+from openerp.addons.website_sale.controllers.main import website_sale
+
+
+class WebsiteSale(website_sale):
+
+    def _get_vat_code(self, code=None):
+        if code:
+            return request.env['res.country'].search([
+                ('code', '=', code)
+            ])
+        return (request.website.vat_code_id or
+                request.website.company_id.country_id)
+
+    @http.route(
+        ['/shop/confirm_order'], type='http', auth="public", website=True)
```

Comment:
No need to pass all parameters to `route` if the method was already routed in parent controller. Just put `@http.route()` and all is inherited.